source/server/listener_manager_impl.cc

```diff
@@ -197,8 +197,14 @@ ListenerImpl::ListenerImpl(const envoy::api::v2::Listener& config, const std::st
     ProtobufTypes::MessagePtr message =
         Config::Utility::translateToFactoryConfig(transport_socket, config_factory);
 
-    std::vector<std::string> server_names(filter_chain_match.sni_domains().begin(),
-                                          filter_chain_match.sni_domains().end());
+    std::vector<std::string> server_names;
```

Comment:
I think we should throw an exception if both sni_domains and server_names are non-empty (and document the behavior).

Reply:
Done.